onnxruntime/core/providers/cpu/optional/optional_ops.cc

```diff
@@ -118,7 +118,11 @@ Status Optional::Compute(OpKernelContext* ctx) const {
 
   } else {  // No input was provided - we use the type proto to construct the output OrtValue
 
-    CheckValidTypeProto(*type_proto_);
+    if (!CheckValidTypeProto(*type_proto_)) {
+      return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
+                             "The TypeProto attribute in the Optional op ",
+                             "can only be of type(tensor) or (seq(tensor))");
+    }
 
```

Comment:
Does the ONNX validation (either type/shape inferencing or the type constraints for the operator) not check this?

Reply:
The type constraints for the operator can't check this as this is an attribute (which is a generic TypeProto). The type/shape inferencing seems to just copy over the provided tp as the output tp when it probably should have also checked to ensure that the provided tp is a tensor or a seq(tensor) given that the output type of the op can only be those types - https://github.com/onnx/onnx/blob/805ae1e634697e37b43701e585c9c253a29ce076/onnx/defs/optional/defs.cc#L53.